plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -89,6 +95,16 @@ import {
   IdentityApiGetIdentityRequest,
 } from '@backstage/plugin-auth-node';
 import { InternalTaskSecrets } from '../scaffolder/tasks/types';
+import { checkPermission } from '../util/checkPermissions';
+
+/**
+ *
+ * @public
+ */
+export type ScaffolderPermissionRuleInput =
```

Comment:
When I try to keep it as an internal type and not export it, the `api-report.md` will contain the following warning:
`  // Warning: (ae-forgotten-export) The symbol "ScaffolderPermissionRuleInput" needs to be exported by the entry point index.d.ts
`

Reply:
When setting it to an `@internal` like follows:
```ts
/**
 * 
 * @internal
 */
type ScaffolderPermissionRuleInput =
  | TemplatePermissionRuleInput
  | ActionPermissionRuleInput;
```
Running `yarn build:api-docs plugins/scaffolder-backend` throws the following error:
```
# Compiling TypeScript
# Generating package API reports
## Processing plugins/scaffolder-backend

Internal Error: Unable to follow symbol for "ScaffolderPermissionRuleInput"

You have encountered a software defect. Please consider reporting the issue to the maintainers of this application.
```

Reply:
Any ideas on how to make not export this and have the `api-report.md` not have warnings?

Reply:
I wonder if it's because it's marked as `@internal` but it's actually depended on by ` @public` type? If it's part of the public API then it will need to be `@public` I think.

Reply:
yeah if you rollback the following the warning should go away

```diff
-  permissionRules?: Array<ScaffolderPermissionRuleInput>;
+  permissionRules?: Array<
+    TemplatePermissionRuleInput | ActionPermissionRuleInput
+  >;
```

Reply:
Reverted in https://github.com/backstage/backstage/pull/24518/commits/cb2ed9736705ba6aa815782c47116aa243b195b8